addons/mail/static/src/models/composer/composer.js

```diff
@@ -367,6 +372,63 @@ function factory(dependencies) {
             }
         }
 
+        updateMentions() {
+            const channels = [];
+            const partners = [];
+            const div = document.createElement('div');
+            div.innerHTML = this.message.body;
+            const anchors = div.querySelectorAll('a');
+
+            anchors.forEach((element) => {
+                const id = parseInt(element.dataset.oeId);
+                if (element.dataset.oeModel === 'res.partner') {
+                    const partner = this.env.models["mail.partner"].find(partner => partner.id === id);
+                    if (partner) {
+                        partners.push(partner);
+                    }
+                } else if (element.dataset.oeModel === 'mail.channel') {
+                    const channel = this.env.models["mail.thread"].find(channel => channel.id === id);
+                    if (channel) {
+                        channels.push(channel);
+                    }
+                }
+            });
+            this.update({
+                mentionedPartners: [['link', partners]],
+                mentionedChannels: [['link', channels]],
+            });
+        }
+
+        async updateMessage() {
+            const attachmentIds = this.attachments.map((attachment) => attachment.id);
+            if (
+                htmlToTextContentInline(this.message.body) === this.textInputContent &&
+                JSON.stringify(attachmentIds) === JSON.stringify(this.messageAttachments)
+            ) {
+                this.message.update({ isEditingMessage: false });
+                return;
+            }
+            this.updateMentions();
+            const vals = {
+                body: this.getBody(),
+                attachment_ids: attachmentIds,
+            };
+            const [messageData] = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'update_message',
+                args: [[this.message.id], vals],
+            }));
+
+            const data = this.env.models['mail.message'].convertData(messageData);
+            this.message.update(Object.assign(data, { isEditingMessage: false }));
+            if (this.thread) {
+                this.update({
+                    textInputContent: "",
+                    isLastStateChangeProgrammatic: true,
+                });
+            }
+        }
```

Comment:
I don't like that we need to handle this manually. I still think we should postpone this task until we have the new web editor ready to use.